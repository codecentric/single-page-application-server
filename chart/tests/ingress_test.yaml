suite: test ingress
templates:
  - ingress.yaml
tests:
  - it: should not render ingress by default
    asserts:
      - hasDocuments:
          count: 0
  - it: should fail to render if enabled but className is not defined
    values:
      - ./values/minimal_values.yaml
    set:
      ingress:
        enabled: true
      http:
        hosts:
          "test.example.com": {}
    asserts:
      - failedTemplate:
          errorMessage: "ingress.className is required!"
  - it: should fail to render if enabled but http.hosts are not defined
    values:
      - ./values/minimal_values.yaml
    set:
      ingress:
        enabled: true
        className: "testingress"
    asserts:
      - failedTemplate:
          errorMessage: "http.hosts is required if ingress is enabled!"
  - it: should render ingress with default values if enabled
    values:
      - ./values/minimal_values.yaml
    set:
      ingress:
        enabled: true
        className: "testingress"
      http:
        hosts:
          "test.example.com": {}
    asserts:
      - matchSnapshot: { }
  - it: should render ingress with default.base_href if path is not defined
    values:
      - ./values/minimal_values.yaml
    set:
      config.default.base_href: "/base-path"
      ingress:
        enabled: true
        className: "testingress"
      http:
        hosts:
          "test.example.com": { }
    asserts:
      - matchSnapshot: { }
  - it: should render host with custom path and targetServicePortName
    values:
      - ./values/minimal_values.yaml
    set:
      ingress:
        enabled: true
        className: "testingress"
      http:
        hosts:
          "test.example.com":
            path: /custom-path
            targetServicePortName: https
    asserts:
      - matchSnapshot: { }
  - it: should render tls property if http."host".tlsSecretName is set
    values:
      - ./values/minimal_values.yaml
    set:
      ingress:
        enabled: true
        className: "testingress"
      http:
        hosts:
          "test.example.com":
            tlsSecretName: my-tls-secret
    asserts:
      - matchSnapshot: { }
