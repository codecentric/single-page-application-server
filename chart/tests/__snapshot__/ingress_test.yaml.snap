should render host with custom path and targetServicePortName:
  1: |
    apiVersion: networking.k8s.io/v1
    kind: Ingress
    metadata:
      labels:
        app.kubernetes.io/component: frontend
        app.kubernetes.io/instance: default
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: RELEASE-NAME
        app.kubernetes.io/version: latest
        helm.sh/chart: single-page-application-server-chart-0.0.0
      name: RELEASE-NAME
    spec:
      ingressClassName: testingress
      rules:
        - host: test.example.com
          http:
            paths:
              - backend:
                  service:
                    name: RELEASE-NAME
                    port:
                      name: http
                path: /custom-path
                pathType: Prefix
      tls: null
should render ingress with default values if enabled:
  1: |
    apiVersion: networking.k8s.io/v1
    kind: Ingress
    metadata:
      labels:
        app.kubernetes.io/component: frontend
        app.kubernetes.io/instance: default
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: RELEASE-NAME
        app.kubernetes.io/version: latest
        helm.sh/chart: single-page-application-server-chart-0.0.0
      name: RELEASE-NAME
    spec:
      ingressClassName: testingress
      rules:
        - host: test.example.com
          http:
            paths:
              - backend:
                  service:
                    name: RELEASE-NAME
                    port:
                      name: http
                path: /
                pathType: Prefix
      tls: null
should render ingress with default.base_href if path is not defined:
  1: |
    apiVersion: networking.k8s.io/v1
    kind: Ingress
    metadata:
      labels:
        app.kubernetes.io/component: frontend
        app.kubernetes.io/instance: default
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: RELEASE-NAME
        app.kubernetes.io/version: latest
        helm.sh/chart: single-page-application-server-chart-0.0.0
      name: RELEASE-NAME
    spec:
      ingressClassName: testingress
      rules:
        - host: test.example.com
          http:
            paths:
              - backend:
                  service:
                    name: RELEASE-NAME
                    port:
                      name: http
                path: /base-path
                pathType: Prefix
      tls: null
should render tls property if http."host".tlsSecretName is set:
  1: |
    apiVersion: networking.k8s.io/v1
    kind: Ingress
    metadata:
      labels:
        app.kubernetes.io/component: frontend
        app.kubernetes.io/instance: default
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: RELEASE-NAME
        app.kubernetes.io/version: latest
        helm.sh/chart: single-page-application-server-chart-0.0.0
      name: RELEASE-NAME
    spec:
      ingressClassName: testingress
      rules:
        - host: test.example.com
          http:
            paths:
              - backend:
                  service:
                    name: RELEASE-NAME
                    port:
                      name: http
                path: /
                pathType: Prefix
      tls:
        - hosts:
            - test.example.com
          secretName: my-tls-secret
